libfreerdp/core/redirection.c

```diff
@@ -413,16 +410,15 @@ static BOOL rdp_redirection_write_target_cert(wStream* s, const rdpRedirection*
 static BOOL rdp_redireciton_write_target_cert_stream(wStream* s, const rdpRedirection* redirection)
 {
 	BOOL rc = FALSE;
-	wStream* serialized = Stream_New(NULL, 1024);
+	wStream* serialized = Stream_New(NULL, 2048);
 	if (!serialized)
 		goto fail;
 
 	if (!rdp_redirection_write_target_cert(serialized, redirection))
 		goto fail;
 
-	if (!rdp_redirection_write_base64_wchar(
-	        LB_TARGET_CERTIFICATE, s, Stream_GetPosition(serialized), Stream_Buffer(serialized)))
-		return FALSE;
+	rc = rdp_redirection_write_base64_wchar(
```

Comment:
`rdp_redirection_write_base64_wchar` should take a `wStream` as argument that would allow check for `serialized`'s free space (otherwise how can we ensure that 2048 is enough ?)

Reply:
@hardening `rdp_redirection_write_target_cert` writes to `serialized`, `rdp_redierction_write_base64_wchar` writes to `s` (and takes the data from `serialized`)